### PR TITLE
Add view all precinct tally report button

### PR DESCRIPTION
**Pull request by carolinemodic**
Adds a button to print all the tally reports by precinct at once. I put this in a new section for default tally reports and also moved the overall tally report button there. I plan to eventually add a "Create custom tally report" button or buttons for any other default reports we want to this section. 

In printing the full precinct by precinct tallies I also noticed a bug from my last PR where I wasn't initializing empty tallies for each precinct when there was no data which caused the precinct reports to be totally blank instead of showing zero tallies. I also fixed that bug and added a test for it. 

### Tally Screen
Screen Shot 2021-02-01 at 12 28 32 PM
(scrolled further) 
Screen Shot 2021-02-01 at 12 28 38 PM

### Tally Report Screen for printing all precincts tally report 
Screen Shot 2021-01-29 at 4 43 35 PM

### CODE REVIEW

File: apps/election-manager/src/screens/TallyScreen.tsx [hunk 1]
```diff
@@ -134,6 +134,21 @@ const TallyScreen: React.FC = () => {
     <Loading>Tabulating Results…</Loading>
   ) : (
     <React.Fragment>
+      <h2>{statusPrefix} Tally Reports</h2>
+      <p>
+        <LinkButton to={routerPaths.tallyFullReport}>
+          {hasCastVoteRecordFiles
+            ? 'View Full Election Tally Report'
+            : 'View Zero Tally Report'}
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyScreen.tsx:142]:
```suggestion
          View {statusPrefix} Full Election Tally
```

File: apps/election-manager/src/screens/TallyScreen.tsx [hunk 2]
```diff
@@ -134,6 +134,21 @@ const TallyScreen: React.FC = () => {
     <Loading>Tabulating Results…</Loading>
   ) : (
     <React.Fragment>
+      <h2>{statusPrefix} Tally Reports</h2>
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyScreen.tsx:137]:
```suggestion
      <h2>Tally Reports</h2>
```

**Review by beausmith — CHANGES_REQUESTED** [review]
I'm concerned that moving the location of the tally report buttons above the table of results, changes the flow of the page sections by heading. Before it was 1, 2, 3… now it's 1, 3, 2. 

I suggest that this commit be soft reset and then separate commits made only for the fix and the feature without the reorder of sections or update to button label text:

- fix(election-manager): Initialize empty tallies for each precinct
- feat(election-manager): Print all precinct reports together

**Review by benadida — APPROVED** [review]
I agree with feedback on not hard-wiring even the text of "zero report", otherwise nothing more to add. I'm approving this cause nothing more to add, leaving conversation of layout to @carolinemodic and @beausmith

**Comment by carolinemodic** [comment]
back to you @beausmith , updated the description with new screenshots. Good nudge on breaking up to two commits!

File: apps/election-manager/src/screens/TallyReportScreen.tsx [hunk 3]
```diff
@@ -85,6 +95,9 @@ const TallyReportScreen: React.FC = () => {
     if (scannerId) {
       return `${statusPrefix} Scanner Tally Report for Scanner ${scannerId}`
     }
+    if (precinctIdFromProps === 'all') {
+      return `${statusPrefix} ${election.title} All Precincts Tally Report`
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyReportScreen.tsx:99]:
```suggestion
      return `${statusPrefix} ${election.title} Tally Reports for All Precincts`
```

File: apps/election-manager/src/App.test.tsx [hunk 4]
```diff
@@ -199,28 +199,56 @@ it('tabulating CVRs', async () => {
...
   expect(getByTestId('tally-report-contents')).toMatchSnapshot()
+
+  fireEvent.click(getByText('Tally'))
+
+  await waitFor(() => {
+    fireEvent.click(getByText('View Official Tally Reports For All Precincts'))
+  })
+
+  getByText(
+    'Official Mock General Election Choctaw 2020 All Precincts Tally Report'
```
Comment (beausmith) [inline apps/election-manager/src/App.test.tsx:236]:
```suggestion
    'Official Mock General Election Choctaw 2020 Tally Reports for All Precincts'
```

File: apps/election-manager/src/screens/TallyScreen.tsx [hunk 5]
```diff
@@ -184,8 +184,13 @@ const TallyScreen: React.FC = () => {
...
             <TD>
-              <LinkButton to={routerPaths.tallyFullReport}>
-                View {statusPrefix} Full Election Tally
+              <LinkButton
+                small
+                to={routerPaths.tallyPrecinctReport({
+                  precinctId: 'all',
+                })}
+              >
+                View {statusPrefix} Tally Reports For All Precincts
```
Comment (beausmith) [inline apps/election-manager/src/screens/TallyScreen.tsx:193]:
```suggestion
                View {statusPrefix} Tally Reports for All Precincts
```

File: apps/election-manager/src/App.test.tsx [hunk 6]
```diff
@@ -199,28 +199,56 @@ it('tabulating CVRs', async () => {
...
   expect(
     getAllByText('Official Mock General Election Choctaw 2020 Tally Report')
       .length > 0
   ).toBe(true)
   expect(getByTestId('tally-report-contents')).toMatchSnapshot()
+
+  fireEvent.click(getByText('Tally'))
+
+  await waitFor(() => {
+    fireEvent.click(getByText('View Official Tally Reports For All Precincts'))
```
Comment (beausmith) [inline apps/election-manager/src/App.test.tsx:232]:
```suggestion
    fireEvent.click(getByText('View Official Tally Reports for All Precincts'))
```

**Review by beausmith — APPROVED** [review]
LGTM 👍  … with a minor copy consistency change across a few files.

**Pull request merged**